x/programs/cmd/simulator/src/lib.rs

```diff
@@ -236,16 +279,13 @@ impl Client {
         Self { path }
     }
 
-    /// Runs a [Plan] against the simulator and returns vec of result.
-    /// # Errors
-    ///
-    /// Returns an error if the serialization or plan fails.
-    pub fn run_plan(self, plan: Plan) -> Result<Vec<PlanResponse>, ClientError> {
-        let Child {
-            mut stdin,
-            mut stdout,
-            ..
-        } = Command::new(self.path)
+    pub fn try_build(
+        self,
+    ) -> Result<
+        Client<impl Write, impl Iterator<Item = Result<PlanResponse, ClientError>>>,
+        ClientError,
+    > {
```

Comment:
(please do this before I approve)

Reply:
Because of the last changes, I just opted for a simpler alternative of creating a new type over `Result`
```rust
type StepResult = Result<PlanResponse, StepError>;
```

Reply:
@richardpringle are you okay with that choice ? I'm wondering if you made the iteration response generic to allow the transport layer to be generic, but I think that we should do that later